pkg/scheduler/queue/tree/tenant_querier_queuing_algorithm.go

```diff
@@ -222,27 +224,43 @@ func (qa *TenantQuerierQueuingAlgorithm) dequeueUpdateState(node *Node, dequeued
 	// delete from shared tenantNodes
 	for i, tenantNode := range qa.tenantNodes[childName] {
 		if tenantNode == dequeuedFrom {
-			qa.tenantNodes[childName] = append(qa.tenantNodes[childName][:i], qa.tenantNodes[childName][i+1:]...)
+			qa.tenantNodes[childName] = slices.Delete(qa.tenantNodes[childName], i, i+1)
```

Comment:
not related to the perf issue but using slices.Delete in the other code showed a slight improvement over these append-spread approaches and is 100x easier to understand.

Reply:
To clarify: did you benchmark this change as well? I'd expect these two cases to perform slightly differently given one is removing an element from the middle of the slice and the one below is removing elements from the end of the slice.

Reply:
no this change was not part of the benchmark - this is not a behavior change, just updating to use the new slices built-ins for the same purpose as before

Reply:
Could you please expand the benchmark to include this change? I'm not expecting a dramatic difference due to this change, but it'd be good to confirm.

Reply:
I’ll need to create a separate benghmark suite since this is not a hot path. the slice never exceeds length 4.
Is this necessary?